python/revng/cli/_commands/idb_converter.py

```diff
@@ -0,0 +1,655 @@
+                except Exception as exception:
+                    log_error(f"Unable to parse function type for {function_name}.")
+                    log_error(str(exception))
+                    function_type = None
+                    return True
+
+                revng_function_type = None
+                if function_type is not None:
+                    qualified_revng_function_type = self._convert_idb_type_to_revng_type(
+                        function_type
+                    )
+                    revng_function_type = self.unwrap_qualified(qualified_revng_function_type)
+                else:
+                    abi = revng_arch_to_abiname[self.arch]
+
+                    qualified_return_type = self._get_primitive_type(m.PrimitiveTypeKind.Void, 0)
+                    revng_function_type = m.CABIFunctionType(
+                        OriginalName=function_name,
+                        ABI=abi,
+                        ReturnType=qualified_return_type,
+                        Arguments=[],
+                    )
+                    self.revng_types_by_id[revng_function_type.ID] = revng_function_type
+
+                dynamic_function = m.DynamicFunction(
+                    OriginalName=function_name,
+                    Prototype=m.Reference.create(m.Binary, revng_function_type),
+                )
+                self.dynamic_functions.append(dynamic_function)
+                return True
+
+            mod_name = self.api.ida_nalt.get_import_module_name(mod_index)
+            # NOTE: IDA 7 has a bug when reporting imported library names - prints `.dynsym` for
+            # each import.
+            if mod_name and mod_name != ".dynsym":
+                self.imported_libraries.append(mod_name)
+            self.api.ida_nalt.enum_import_names(mod_index, cb)
+
+    def _fixup_void_cabifunctiontype_arguments(self):
+        """Fixes CABIFunctionType objects with arguments of type
+        void (functions with prototype `ret_type f(void, ...)`)
+        by replacing the void argument with a generic primitive type.
+        """
+        for type_id, t in self.revng_types_by_id.items():
+            if not isinstance(t, m.CABIFunctionType):
+                continue
+
+            for argument in t.Arguments:
+                if len(argument.Type.Qualifiers) > 0:
+                    continue
+
+                argument_unqualified_type = self.resolve_typeref(argument.Type.UnqualifiedType)
+
+                if not argument_unqualified_type:
+                    continue
+
+                if not isinstance(argument_unqualified_type, m.PrimitiveType):
+                    continue
+
+                if argument_unqualified_type.PrimitiveKind != m.PrimitiveTypeKind.Void:
+                    continue
+
+                # TODO: Pick appropriate size.
+                argument.Type = self._get_primitive_type(m.PrimitiveTypeKind.Generic, 8)
+
+    def _fixup_structs(self):
```

Comment:
In this function there's *a lot* of logic for bitfields. We could ignore them, but then, since IIRC we compute the size of a IDB's `struct` as the sum of the size of all of its members, we'd lose the entirety of the `struct`.
Please review this logic and test it.
I don't feel super confident about it.

Reply:
O yeah... ignoring the whole struct would be more appropriate way, right?

Reply:
Wait, who is "it"?
What would be `size(my_struct)` in C?
What does IDA say from the UI?
What does our code say?

We either need to investigate more or drop. Up to you.

Reply:
Sorry for not being super clear.

The `sizeof`, as well as both `revng import binary` and IDA, are accurate - both report `8`.

Our code (the code from the original PR) implements the logic in the wrong way. I'd say that it is not trivial to implement it with the current information (I think we would need to improve/investigate the `python-idb` package to correctly calculate/pick the size of the struct from IDB files). Therefore I dropped this for now.

Reply:
Gotcha, leave a TODO and let's call it a day.